### ISSUE TITLE
Don't include properties that are not required in the value

### ISSUE THREAD

**Issue by mohsen1** [jdorn/json-editor#406]
With this schema:

```json
{
  "type": "object",
  "properties": {
    "name": {
      "type": "string"
    },
    "age": {
      "type": "integer"
    }
  }
}
```
If user doesn't put any data in, the output value should be `{}`. None of the properties are required.

If a property is required, then it makes sense to fill in the value for required property with some default value:

```json
{
  "type": "object",
  "properties": {
    "name": {
      "type": "string"
    },
    "age": {
      "type": "integer"
    }
  },
  "required": ["age"]
}
```

This should output `{"age": 0}` since `"age"` is required and empty object will not conform to the schema

**Comment by stevage** [comment]
There seems to be an undocumented option, `remove_empty_properties: true` that basically does this.

**Comment by mohsen1** [comment]
@stevage Thank you very much! That saved me a lot of work. It would be nice if `remove_empty_properties ` is true then it will do what it was doing for required properties.

**Comment by stevage** [comment]
Yeah. Btw, I was slightly wrong - it's not "undocumented", but it's documented as an editor-level property (not for the whole schema).

**Comment by stevage** [comment]
Also, now that I look at this more closely, I think there is a bug:

```
{
  "type": "object",
  "options": { "remove_empty_properties": true} ,
  "required": [ "age" ],
  "properties": {
    "name": {
      "type": "string"
    },
    "age": {
      "type": "integer"
    }
  }
}
```

This doesn't let you use an age of `0` because it gets removed, even though it's `required`.

**Comment by mohsen1** [comment]
Right, https://github.com/swagger-api/swagger-editor/issues/482 is exactly this issue

**Comment by centromere** [comment]
Any update on this?

**Comment by bozkurt30** [comment]
Hello; For C# (.net 6 ), when you correct your bool value as follows, the false value swagger also works.
public bool? IsBlock { get; set; } = false; "worked after adding a question mark"

good work
